src/ring_theory/polynomial/basic.lean

```diff
@@ -651,7 +651,70 @@ is_noetherian_submodule_left.1 (is_noetherian_of_fg_of_noetherian _
 
 end ideal
 
+section prime
+variables (σ) {r : R}
+
+namespace polynomial
+lemma prime_C_iff : prime r ↔ prime (polynomial.C r) :=
+begin
+  split,
+  { intro hr, have := hr.1, rw ← ideal.span_singleton_prime at hr ⊢,
+    { convert ideal.is_prime_map_C_of_is_prime hr using 1,
+      rw [ideal.map_span, set.image_singleton] },
+      exacts [λ h, this (polynomial.C_eq_zero.1 h), this] },
+  { exact λ hC, ⟨ λ h, hC.1 $ by rw [h, C_0], λ h, hC.2.1 $ h.map C,
+      λ a b h, by { apply (hC.2.2 (C a) (C b) (by { convert C.map_dvd h, simp })).imp _ _;
+        { rw C_dvd_iff_dvd_coeff, intro h, convert h 0, simp } } ⟩ },
+end
+end polynomial
+
+namespace mv_polynomial
+
+private lemma prime_C_iff_of_fintype [fintype σ] : prime r ↔ prime (C r : mv_polynomial σ R) :=
```

Comment:
One more assumption than `prime_C_iff` below, so marked `private`.

Reply:
I understand this is `private`. But I would also swap the sides of the `iff` here.